docs/DeveloperGuide.md

```diff
@@ -1,34 +1,255 @@
+![img.png](Overview%20for%20Safe%20Feature.png)
+Given below is an example usage scenario and how the save mechanism behaves at each step. <br />
+Step 1. The user launches the application for the first time. 
+`AliasStorage`, `HistoryRouteStorage`, `NotesStorage`, `DailyRouteStorage` and `FavouriteLocationsStorage` 
+will be initialized with the respective file paths of `aliasList`,  `historyList`, `notesList`, `dailyRouteList` and `favouritesList`. 
+The lists will be initialised by calling `AliasStorage#loadAlias()`, `HistoryRouteStorage#loadHistory()`, `NotesStorage#loadNotes()` `DailyRouteStorage#loadDailyRoute()` and `FavouriteLocationsStorage#loadFavourites()` with the initial state of the application. <br /> 
+This is done only once for each time the application is launched. <br />
+![img.png](SaveStep1.png)
+<br />
+Step 2. The user executes `go` command to show the route from starting location to final location. <br /> 
+The `go` command calls `HistoryRouteStorage#overwriteHistoryListFile()`, 
+causing the modified state of the `historyList` in the application after the `go` command executes to be saved in the `routesHistoryList.txt`. <br />
+Step 3. The user executes `add note E4/...` to tag a note to that location. <br /> 
+The `add note` command calls `NotesStorage#overwriteNotesListFile()`, causing  the modified state of the `notesList` to be saved into the `notesList.txt`. <br />
+Step 4. The user executes `delete note E4/1` to remove a note with the given note index from that location, assuming that it exists. <br /> 
+The `delete note` command also calls `NotesStorage#overwriteNotesListFile()`, causing  the modified state of the `notesList` to be saved into the `notesList.txt`. <br />
+Step 5. The user executes `like E4` command to add a location to favourites. <br /> The `like` command calls `FavouriteLocationsStorage#save()`, causing the modified state of the `favouritesList` to be saved into the `favouritesList.txt`. <br />
+Step 6. At any point when a command is called, the `AliasStorage#overwriteAliasListFile()`, `HistoryRouteStorage#overwriteHistoryListFile()`, `NotesStorage#overwriteNotesListFile()`, `DailyRouteStorage#overwriteDailyRouteFile()` and `FavouriteLocationsStorage#overwriteFavouritesListFile()` methods will be executed, 
+but not all files will be modified. The above steps explains which lists will be modified after the commands listed above are called.
+For all other commands, they also call the overwrite functions, but they do not modify the state of any of the lists `aliasList`,  `historyList`, `notesList`,  `dailyRouteList` and `favouritesList`. 
+Thus, the `aliasList.txt`, `routesHistoryList.txt`, `notesList.txt`, `dailyRouteList.txt`  and `favouritesList.txt` inside the created `data` folder remains unchanged. <br/>
+#### Design Consideration
+Alternative 1 (current choice): Saves the entire list of block aliases, visited routes, tagged notes, daily routes and favourite locations. <br/>
```

Comment:
![image](https://user-images.githubusercontent.com/77423978/113076789-69f22980-9202-11eb-90b5-28a914ea6e56.png)

One choice is not an alternative choice. Maybe you can phrase it to it being just, "Considerations taken" and list down what was considered when implementing.

Reply:
Maybe list the other choices if you want to keep the current format